src/software/jetson_nano/thunderloop.cpp

```diff
@@ -35,37 +91,98 @@ Thunderloop::~Thunderloop()
 
 /*
  * Run the main robot loop!
- *
- * @param The rate to run the loop
  */
-void Thunderloop::run(unsigned run_at_hz)
+void Thunderloop::runLoop()
 {
-    using clock = std::chrono::steady_clock;
+    // Timing
+    struct timespec next_shot;
+    struct timespec start_time;
+    struct timespec end_time;
+    struct timespec time;
+
+#define TIME_EXPRESSION(arg)                                                             \
+    clock_gettime(0, &start_time);                                                       \
+    arg;                                                                                 \
+    clock_gettime(0, &end_time);                                                         \
+    timespec_diff(&end_time, &start_time, &time);
 
```

Comment:
doesn't seem like this is recording the time?

Reply:
it stores it in time